lib/src/uri_template.dart

```diff
@@ -61,8 +62,10 @@ class UriParser extends UriPattern {
   RegExp get fragmentRegex => _fragmentRegex;
   RegExp get pathRegex => _pathRegex;
 
-  UriParser(UriTemplate this.template, {bool fragmentPrefixMatching: true})
-      : _fragmentPrefixMatching = firstNonNull(fragmentPrefixMatching, true) {
+  UriParser(UriTemplate this.template, {bool fragmentPrefixMatching: true,
+    bool queryParamsAreOptional: false})
```

Comment:
style nit: when wrapping named parameters I like to just put every one on a new line